core/app/models/spree/promotion_handler/coupon.rb

```diff
@@ -13,7 +13,7 @@ def apply
           if promotion.present? && promotion.actions.exists?
             handle_present_promotion(promotion)
           else
-            if promotion_code && promotion_code.promotion.expired?
+            if promotion.present? && promotion.expired?
```

Comment:
hmm, probably it won't work. `promotion.active?` should return false when it's expired. I will fix that...